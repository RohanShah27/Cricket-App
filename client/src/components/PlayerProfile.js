import React, { Component } from "react";
import Navigation from "./Navigation";
import "../styles/playerProfile.css";
import { connect } from "react-redux";
import { searchPlayer } from "../actions/Players";
import virat from "../virat-kohli.png";

export class PlayerProfile extends Component {
  constructor(props) {
    // super(props) -> its parents and all of its parents properties
    super(props);
    // this.searchForPlayer = this.searchForPlayer.bind(this);
  }
  componentDidMount() {
    console.log(this.props);
    let player = {
      playerName: this.props.match.params.playerName
    };
    this.props.searchPlayer(player);
    console.log(this.props);
  }
  render() {
    console.log(this.props.player[0].batting_style);
    return (
      <div>
        <div className="main-section">
          <div className="flex-container">
            <div style={{ flexGrow: 1 }}>
              <img src={virat}></img>
            </div>
            <div style={{ flexGrow: 8 }}>
              <h1>{this.props.player[0].player_name}</h1>
              <h5>Team: {this.props.player[0].player_nation}</h5>
              <div className="profile">
                Passionate. No word describes Virat Kohli better. His passion
                for cricket has made him one of the best batsmen in the world
                across formats, and has also helped him grow into a ruthless
                captain. It's also passion that defines Kohli's emotional,
                effervescent and at times firecracker character. Virat Kohli
                does not hold back and that remains his strength.
              </div>
            </div>
          </div>
          <div className="row">
            <div className="side">
              <h2>Personal Details</h2>
              <p>Date Of Birth: {this.props.player[0].player_dob}</p>
              <p>Batting Style: {this.props.player[0].batting_style}</p>
              <p>Bowling Style: {this.props.player[0].bowling_style}</p>
              <p>Role: {this.props.player[0].player_role}</p>
            </div>
            <div className="main">
              <h2>Player Stats</h2>
              <div className="wrapper">
                <div className="box-a">
                  <b>ODI Ranking</b>
                </div>
                <div className="box-b">
                  <b>Test Ranking</b>
                </div>
                <div className="box-c">
                  <b>T20 Ranking</b>
                </div>
                <div className="stats-row">
                  <div className="box-d">
                    {this.props.player[0].odi_ranking}
                  </div>
                  <div className="box-e">
                    {this.props.player[0].test_ranking}
                  </div>
                  <div className="box-f">
                    {this.props.player[0].t20_ranking}
                  </div>
                </div>
              </div>
            </div>
          </div>
          {/* <div className="image-container"></div>
          <div className="playerdetails"></div>
          <div className="personalinfo"> </div>

          <div className="playerstats">
            
          </div>
          <div className="graphs"></div> */}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  player: state.playerReducer.player
});
export default connect(
  mapStateToProps,
  { searchPlayer }
)(PlayerProfile);
