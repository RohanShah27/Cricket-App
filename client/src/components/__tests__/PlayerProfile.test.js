import React from "react";
import { shallow, mount } from "enzyme";
import { PlayerProfile } from "../PlayerProfile";
let playerProfile = jest.fn();
let match = { params: { player_id: 1 }, isExact: true, path: "", url: "" };
let history = {
  location: {
    state: {
      player: {
        player_name: "test name"
      }
    }
  }
};
let location = {
  state: {
    player: {
      player_id: 1,
      player_name: "test name",
      nation: "test nation",
      player_dob: "1/2/3",
      player_role: "test role",
      batting_style: "test style",
      bowling_style: "test style"
    }
  }
};
let player = {
  ODI: {
    totalRuns: 1,
    totalWickets: 1,
    total50s: 1,
    total4s: 1,
    total6s: 1,
    totalcenturies: 1,
    totaldoublecenturies: 1
  },
  Test: {
    totalRuns: 1,
    totalWickets: 1,
    total50s: 1,
    total4s: 1,
    total6s: 1,
    totalcenturies: 1,
    totaldoublecenturies: 1
  },
  T20: {
    totalRuns: 1,
    totalWickets: 1,
    total50s: 1,
    total4s: 1,
    total6s: 1,
    totalcenturies: 1,
    totaldoublecenturies: 1
  }
};

let searchPlayer = jest.fn();
const wrapper = mount(
  <PlayerProfile
    player={player}
    playerProfile={playerProfile}
    match={match}
    history={history}
    location={location}
    searchPlayer={searchPlayer}
  />
);
describe("Test suite for Player Profile Component", () => {
  it("Should Render the component and match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should display player name", () => {
    expect(wrapper.find("#playerName").props().children).toBe(
      history.location.state.player.player_name
    );
  });
  it("should display player Nation", () => {
    expect(wrapper.find("#playerNation").props().children).toStrictEqual(
      ["Team:", "test nation"]
      //   "Team: " + location.state.player.nation
    );
  });
  it("should display player DOB", () => {
    expect(wrapper.find("#playerDOB").props().children).toStrictEqual([
      " ",
      "1/2/3"
    ]);
  });
  it("should display player Role", () => {
    expect(wrapper.find("#playerRole").props().children).toBe(
      location.state.player.player_role
    );
  });
  it("should display player BattingStyle", () => {
    expect(wrapper.find("#battingStyle").props().children).toBe(
      location.state.player.batting_style
    );
  });
  it("should display player BowlingStyle", () => {
    expect(wrapper.find("#bowlingStyle").props().children).toBe(
      location.state.player.bowling_style
    );
  });
  it("should display player StatsKey for ODI", () => {
    expect(wrapper.find("#keyName0").props().children).toBe("ODI");
  });

  it("should display player StatsKEY for Test", () => {
    expect(wrapper.find("#keyName1").props().children).toBe("Test");
  });
  it("should display player StatsKEy for T20", () => {
    expect(wrapper.find("#keyName2").props().children).toBe("T20");
  });
  it("should check if player stats table is present ", () => {
    expect(wrapper.find("#playerStats").props().children).toBe("Player Stats");
  });
  it("check if player cisualization section is present or nor", () => {
    expect(wrapper.find("playerStatsVisualizationSection")).toBeTruthy();
  });
  it("should check for component will mount as a function", () => {
    const componentWillMount = jest.spyOn(
      PlayerProfile.prototype,
      "componentWillMount"
    );
    wrapper.instance().componentWillMount();
    expect(componentWillMount).toHaveBeenCalled();
    expect(searchPlayer).toBeCalledWith(match.params.player_id);
  });
  it("should check for component will recieve as a function", () => {
    const componentWillReceiveProps = jest.spyOn(
      PlayerProfile.prototype,
      "componentWillReceiveProps"
    );
    let nextProps = {
      player_id: 1
    };
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(componentWillReceiveProps).toBeCalledWith(nextProps);
    expect(searchPlayer).toBeCalledWith(nextProps.player_id);
  });
  it("should test for player image present or not", () => {
    expect(wrapper.find("#playerImage")).toBeTruthy();
  });
});
