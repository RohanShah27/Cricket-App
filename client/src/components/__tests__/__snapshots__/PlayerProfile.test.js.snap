// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Player Profile Component Should Render the component and match snapshot 1`] = `
<PlayerProfile
  history={
    Object {
      "location": Object {
        "state": Object {
          "player": Object {
            "player_name": "test name",
          },
        },
      },
    }
  }
  location={
    Object {
      "state": Object {
        "player": Object {
          "batting_style": "test style",
          "bowling_style": "test style",
          "nation": "test nation",
          "player_dob": "1/2/3",
          "player_id": 1,
          "player_name": "test name",
          "player_role": "test role",
        },
      },
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {
        "player_id": 1,
      },
      "path": "",
      "url": "",
    }
  }
  player={
    Object {
      "ODI": Object {
        "total4s": 1,
        "total50s": 1,
        "total6s": 1,
        "totalRuns": 1,
        "totalWickets": 1,
        "totalcenturies": 1,
        "totaldoublecenturies": 1,
      },
      "T20": Object {
        "total4s": 1,
        "total50s": 1,
        "total6s": 1,
        "totalRuns": 1,
        "totalWickets": 1,
        "totalcenturies": 1,
        "totaldoublecenturies": 1,
      },
      "Test": Object {
        "total4s": 1,
        "total50s": 1,
        "total6s": 1,
        "totalRuns": 1,
        "totalWickets": 1,
        "totalcenturies": 1,
        "totaldoublecenturies": 1,
      },
    }
  }
  playerProfile={[MockFunction]}
  searchPlayer={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "80px",
      }
    }
  >
    <div
      className="main-section"
    >
      <div
        className="player-flex-container"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "flexGrow": 2,
            }
          }
        >
          <img
            className="profile-img"
            id="playerImage"
            src="virat-kohli.png"
          />
        </div>
        <div
          style={
            Object {
              "flexGrow": 8,
            }
          }
        >
          <h1
            id="playerName"
          >
            test name
          </h1>
          <h5
            id="playerNation"
          >
            Team:
            test nation
          </h5>
          <div
            className="profile"
          >
            Passionate. No word describes Virat Kohli better. His passion for cricket has made him one of the best batsmen in the world across formats, and has also helped him grow into a ruthless captain. It's also passion that defines Kohli's emotional, effervescent and at times firecracker character. Virat Kohli does not hold back and that remains his strength.
          </div>
        </div>
      </div>
      <div
        className="player-row"
      >
        <div
          className="side"
        >
          <h2>
            Personal Details
          </h2>
          <p>
            Date Of Birth:
            <br />
            <span
              className="personal-detail-content"
              id="playerDOB"
            >
               
              1/2/3
            </span>
          </p>
          <hr
            className="line"
          />
          <p>
            Role: 
            <br />
            <span
              className="personal-detail-content"
              id="playerRole"
            >
              test role
            </span>
          </p>
          <hr
            className="line"
          />
          <p>
            Batting Style:
            <br />
             
            <span
              className="personal-detail-content"
              id="battingStyle"
            >
              test style
            </span>
          </p>
          <hr
            className="line"
          />
          <p>
            Bowling Style: 
            <br />
            <span
              className="personal-detail-content"
              id="bowlingStyle"
            >
              test style
            </span>
             
          </p>
        </div>
        <div
          className="main"
        >
          <h2
            id="playerStats"
          >
            Player Stats
          </h2>
          <div>
            <table
              className="players-stats-table"
            >
              <tr>
                <th>
                  Format
                </th>
                <th>
                  200's
                </th>
                <th>
                  100's
                </th>
                <th>
                  Runs
                </th>
                <th>
                  6's
                </th>
                <th>
                  4's
                </th>
                <th>
                  Wickets
                </th>
                <th>
                  50's
                </th>
              </tr>
              <tr>
                <td
                  id="keyName0"
                >
                  ODI
                </td>
                <td
                  id="keyValue0"
                >
                  1
                </td>
                <td
                  id="keyValue1"
                >
                  1
                </td>
                <td
                  id="keyValue2"
                >
                  1
                </td>
                <td
                  id="keyValue3"
                >
                  1
                </td>
                <td
                  id="keyValue4"
                >
                  1
                </td>
                <td
                  id="keyValue5"
                >
                  1
                </td>
                <td
                  id="keyValue6"
                >
                  1
                </td>
              </tr>
              <tr>
                <td
                  id="keyName1"
                >
                  Test
                </td>
                <td
                  id="keyValue0"
                >
                  1
                </td>
                <td
                  id="keyValue1"
                >
                  1
                </td>
                <td
                  id="keyValue2"
                >
                  1
                </td>
                <td
                  id="keyValue3"
                >
                  1
                </td>
                <td
                  id="keyValue4"
                >
                  1
                </td>
                <td
                  id="keyValue5"
                >
                  1
                </td>
                <td
                  id="keyValue6"
                >
                  1
                </td>
              </tr>
              <tr>
                <td
                  id="keyName2"
                >
                  T20
                </td>
                <td
                  id="keyValue0"
                >
                  1
                </td>
                <td
                  id="keyValue1"
                >
                  1
                </td>
                <td
                  id="keyValue2"
                >
                  1
                </td>
                <td
                  id="keyValue3"
                >
                  1
                </td>
                <td
                  id="keyValue4"
                >
                  1
                </td>
                <td
                  id="keyValue5"
                >
                  1
                </td>
                <td
                  id="keyValue6"
                >
                  1
                </td>
              </tr>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      id="playerStatsVisualizationSection"
    />
  </div>
</PlayerProfile>
`;
